src/functions-and-arrays.js

```diff
@@ -1,41 +1,162 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
+function maxOfTwoNumbers( a, b) {
+  if (a>b){
+    return a;
+  }
+  else if (a<b){
+    return b;
+  }
+  else{
+    return a;
+  }
+}
 
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
+function findLongestWord(wordArray) {
+  if (wordArray.length === 0){
+    return null;
+  }
+  if (wordArray.length === 1){
+    return wordArray[0];
+  }
+
+  let wordLength=0;
+  let wordOcurrence;
+  for(var i=0; i<= wordArray.length-1; i++){
+    if ( wordArray[i].length > wordLength ){
+      wordLength=wordArray[i].length;
+      wordOcurrence=wordArray[i];
+    }
+  }
+  return wordOcurrence;
+}
 
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
+function sumNumbers(arr) {
+  if (arr.length === 0){
+    return 0;
+  }
+  if (arr.length === 1){
+    return arr[0];
+  }
+  let sum=0;
+  for(var i=0; i<= arr.length-1; i++){
+    sum+=arr[i];
+  }
+  return sum;
+}
 
 
 
 // Iteration #3.1 Bonus:
-function sum() {}
+function sum(arr) {
+  if (arr.length === 0){
+    return 0;
+  }
+  if (arr.length === 1){
+    return arr[0];
+  }
+  let sum=0;
+  for(var i=0; i<= arr.length-1; i++){
+    let arrObjType=typeof(arr[i]);
+    if (arrObjType==="number"){
+      sum+=arr[i];
+    }
+    else if (arrObjType==="string"){
+      sum+=arr[i].length;
+    }
+    else if(arrObjType==="boolean"){
```

Comment:
funnily enough + on a boolean would convert the boolean into a number per default, so you could drop those extra cases entirely. A valid solution would be just handling the string case different and the other two can use the regular + operator and utilize the fact that JS does type coercions on booleans :)